contracts/protocol/modules/LimitOrder.sol

```diff
@@ -64,15 +64,16 @@ abstract contract LimitOrderManager is ClaimableAmountManager {
                 _quantity,
                 openLimitResp.sizeOut
             );
+            (, uint256 marginToVault, uint256 fee) = _positionManager
```

Comment:
instead of always depositing funds and increasing claimableAmount, users just need to deposit funds if they create a new limit order. if sizeOut == quantity means all of the order amounts have worked as a market order and closed current position. if sizeOut < quantity, maybe a part of the limit order worked as a market order. we just need to increase claimableAmount by the margin amount of new limit order (excluded margin of market)

Reply:
How about fill to market and increasing the position?

Reply:
it's the second case, when sizeOut < quantity and a part of the limit order worked as a market order. we handled the increasing of position in function `_openLimitOrder` => `handleMarketPart`. The second case has included increasing and decreasing position